### PR TITLE
Add a variety of credential stores on Linux

### PR DESCRIPTION
**Pull request by mjcheetham**
Add two more credential storage options on Linux.

### CODE REVIEW

File: src/shared/Microsoft.Git.CredentialManager/Interop/Linux/LinuxCredentialStore.cs [hunk 1]
```diff
@@ -45,12 +49,62 @@ public bool Remove(string service, string account)
...
+            string credStoreName = _settings.CredentialBackingStore?.ToLowerInvariant();
+
+            switch (credStoreName)
+            {
+                case "secretservice":
+                    ValidateSecretService();
+                    _backingStore = new SecretServiceCollection(ns);
+                    break;
+
+                case "plaintext":
```
Comment (derrickstolee) [inline src/shared/Microsoft.Git.CredentialManager/Interop/Linux/LinuxCredentialStore.cs:79]: I'm a bit surprised these string literals are not used anywhere else (except the documentation below...). I wonder if it is worth placing them in a constants class?

File: docs/linuxcredstores.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
+# Credential stores on Linux
```
Comment (derrickstolee) [inline docs/linuxcredstores.md:1]: Please add links to this document in the docs for `GCM_CREDENTIAL_STORE` and `credential.credentialStore` for easier discovery.

File: docs/linuxcredstores.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+pin-entry program for the terminal such as `pinentry-tty` or `pinentry-curses`.
+
+If you are connecting to your system via SSH, then the `SSH_TTY` variable should
+automatically be set. GCM Core will pass the value of `SSH_TTY` to GPG/GPG Agent
+as the TTY device to use for prompting for a passphrase.
+
+If you are not connecting via SSH, or otherwise do not have the `SSH_TTY`
+environment variable set, you must set the `GPG_TTY` environment variable before
+running GCM Core. The easiest way to do this is by adding the following to your
+profile (`.bash_rc`, `.profile` etc):
```
Comment (derrickstolee) [inline docs/linuxcredstores.md:81]: Use `~/.bashrc` and `~/.profile`. Note the dropped `_` in `bashrc`.

File: docs/linuxcredstores.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+**:warning: This is not a secure method of credential storage!**
+
+This credential store saves credentials to plaintext files in your file system.
+By default files are stored in `~/.gcm/store` but this can be configured using
+the environment variable `GCM_PLAINTEXT_STORE_PATH` environment variable.
+
+If the directory does not exist is will be created.
+
+**:warning: It is highly recommended you set the permissions on this directory
```
Comment (derrickstolee) [inline docs/linuxcredstores.md:106]: How can we add a liability warning that GCM Core takes no responsibility for the security of this file? We need to HIGHLY recommend the other options.

**Review by derrickstolee — APPROVED** [review]
I've enjoyed using this on my machine, and the documentation is helpful for working through the problems I had.

File: docs/linuxcredstores.md (hunk 4 above)
Comment (mjcheetham) [reply docs/linuxcredstores.md:106]: Time to dust off my Illustrator skills and make a giant ⚠️ symbol, ALL CAPS, and **bold**.

Comment (mjcheetham) [reply docs/linuxcredstores.md]: Updated

File: docs/linuxcredstores.md (hunk 3 above)
Comment (mjcheetham) [reply docs/linuxcredstores.md]: Updated

File: docs/linuxcredstores.md (hunk 2 above)
Comment (mjcheetham) [reply docs/linuxcredstores.md:1]: Done

File: docs/linuxcredstores.md (hunk 4 above)
Comment (mjcheetham) [reply docs/linuxcredstores.md]: Looks like this..

image

File: src/shared/Microsoft.Git.CredentialManager/Interop/Linux/LinuxCredentialStore.cs (hunk 1 above)
Comment (mjcheetham) [reply src/shared/Microsoft.Git.CredentialManager/Interop/Linux/LinuxCredentialStore.cs:79]: I can move them into a constant file, yes.

**Pull request merged**